services/gateway/gateway-service.mjs

```diff
@@ -39,7 +36,7 @@ class GatewayService {
     // A map that tracks the current connected clients.
     // key is the player ID. value is the socket.
     this.socks = {};
-
+    this.moveRule = new MoveRule();
```

Comment:
Don't instantiate this class. They should be utility functions instead of a class member function.
They should be called like `MoveRule.blablaCheck()`.